compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2060,7 +2060,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // struct-like enums (yet...), but it's definitely not
                 // a bug to have constructed one.
                 if adt_kind != AdtKind::Enum {
-                    tcx.check_stability(v_field.did, Some(expr.hir_id), field.span, None);
+                    let field_hir_id = match expr.kind {
+                        hir::ExprKind::Struct(_qpath, fields, _tail) if idx < fields.len() => {
+                            Some(fields[idx].expr.hir_id)
+                        }
+                        _ => None,
+                    };
+                    tcx.check_stability(v_field.did, field_hir_id, field.span, None);
```

Comment:
```suggestion
                    tcx.check_stability(v_field.did, field.hir_id, field.span, None);
```
The field is already available above.